### PR TITLE
chore: align hana plugin

### PR DESCRIPTION
**Pull request by johannes-vogel**

### CODE REVIEW

File: hana/cds-plugin.js [hunk 1]
```diff
@@ -1,6 +1,9 @@
 const cds = require('@sap/cds/lib')
 
+if (!cds.env.fiori.lean_draft) {
+  throw new Error('"@cap-js/hana" only works if cds.fiori.lean_draft is enabled. Please adapt your configuration.')
+}
+
```
Comment (johannes-vogel) [inline hana/cds-plugin.js:6]: Why do we need to set `cds.env.sql.dialect = 'hana'`?

File: hana/package.json [hunk 2]
```diff
@@ -36,9 +36,6 @@
     "requires": {
       "kinds": {
         "sql": {
-          "[development]": {
-            "kind": "sqlite"
```
[the remark this answers is not in the document]
Comment (johannes-vogel) [reply hana/package.json:40]: I've added it now explicitly.

File: hana/cds-plugin.js (hunk 1 above)
Comment (johannes-vogel) [reply hana/cds-plugin.js:6]: Then this is basically "only" required for our tests which generate the plain "CREATE" statements as of now. In production, the deployment is anyway done by HDI?

Comment (johannes-vogel) [reply hana/cds-plugin.js:6]: Yes, but quoted mode is not yet supported, is it?

**Review by BobdenOs — APPROVED** [review]

**Pull request merged**